sdk/search/azure-search/src/samples/java/com/azure/search/SearchOptionsAsyncExample.java

```diff
@@ -0,0 +1,172 @@
+            .endpoint(ENDPOINT)
+            .credential(new SearchApiKeyCredential(API_KEY))
+            .indexName(INDEX_NAME)
+            .buildAsyncClient();
+
+        searchResultsAsList(searchClient);
+        searchResultAsStreamOfPagedResponse(searchClient);
+        searchResultsCountFromStream(searchClient);
+        searchResultsCountFromPage(searchClient);
+        searchResultsCoverage(searchClient);
+        searchResultsCoverageFromPage(searchClient);
+        searchResultsFacetsFromStream(searchClient);
+        searchResultsFacetsFromPage(searchClient);
+    }
+
+    private static void searchResultsFacetsFromPage(SearchIndexAsyncClient searchClient) {
+        // Each page in the response of the search query holds the facets value
+        // Get Facets property from the first page in the response
+        PagedFluxBase<SearchResult, SearchPagedResponse> results = searchClient.search("*",
+            new SearchOptions().setFacets("Rooms/BaseRate,values:5|8|10"),
+            new RequestOptions());
+
+        Map<String, List<FacetResult>> facetResults = results
+            .byPage()
+            .take(1)
+            .map(SearchPagedResponse::getFacets).blockLast();
+
+        facetResults.forEach((k, v) -> {
+            v.forEach(result -> {
+                System.out.println(k + " :");
+                System.out.println("    count: " + result.getCount());
+                result.getDocument().forEach((f, d) -> System.out.println("    " + f + " : " + d));
+            });
+        });
+    }
+
+    private static void searchResultsFacetsFromStream(SearchIndexAsyncClient searchClient) {
+        // Each page in the response of the search query holds the facets value
+        // Accessing Facets property with stream
+        PagedFluxBase<SearchResult, SearchPagedResponse> results = searchClient.search("*",
+            new SearchOptions().setFacets("Rooms/BaseRate,values:5|8|10"),
+            new RequestOptions());
+
+        Stream<Map<String, List<FacetResult>>> facetsMapStream = results
+            .byPage()
+            .map(SearchPagedResponse::getFacets)
+            .toStream();
+
+        facetsMapStream.forEach(result ->
+            result.forEach((k, v) -> v.forEach(facetResult -> {
+                System.out.println(k + " :");
+                System.out.println("    count: " + facetResult.getCount());
+                facetResult.getDocument().forEach((f, d) ->
+                    System.out.println("    " + f + " : " + d)
+                );
+            }))
+        );
+    }
+
+    private static void searchResultsCoverageFromPage(SearchIndexAsyncClient searchClient) {
+        // Each page in the response of the search query holds the coverage value
+        // Get Coverage property from the first page in the response
+        PagedFluxBase<SearchResult, SearchPagedResponse> results = searchClient.search("*",
+            new SearchOptions().setMinimumCoverage(73.5),
```

Comment:
A very minor issue -- In practice, customers would typically pick a round number like 80 for coverage. 73.5 seems oddly precise.